various tests fails sometimes due of parser error

**Describe the bug**
https://github.com/untillpro/airs-bp3/actions/runs/6431021057/job/17466884218
various tests that uses Own VIT Config fails sometimes with errors caused by parser
E.g. `TestBasicUsage_PaymentMethodsManagement()`:
```
--- FAIL: TestBasicUsage_PaymentMethodsManagement (0.42s)
panic: c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:493:2: countries undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:502:2: untill_group_users undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:532:2: untill_user_wages type not supported
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:535:3: report_groups undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3141:2: untill_users undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3143:2: courses undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3144:2: clients undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3162:2: untill_users undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3165:2: untill_users undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3168:2: articles undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3175:2: prices undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3178:2: discount_reasons undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3182:2: bo_service_charge undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3184:2: t2o_groups undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3196:2: order_type undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3200:2: callers undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3201:2: serving_time undefined [recovered]
	panic: c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:493:2: countries undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:502:2: untill_group_users undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:532:2: untill_user_wages type not supported
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:535:3: report_groups undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3141:2: untill_users undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3143:2: courses undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3144:2: clients undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3162:2: untill_users undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3165:2: untill_users undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3168:2: articles undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3175:2: prices undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3178:2: discount_reasons undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3182:2: bo_service_charge undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3184:2: t2o_groups undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3196:2: order_type undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3200:2: callers undefined
c:/workspace/go/airs-bp3/packages/untill/provide.go/schema.sql:3201:2: serving_time undefined

goroutine 14477 [running]:
testing.tRunner.func1.2({0x1354540, 0xc007ec07f8})
	c:/workspace/go/goinstall/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
	c:/workspace/go/goinstall/src/testing/testing.go:1548 +0x397
panic({0x1354540?, 0xc007ec07f8?})
	c:/workspace/go/goinstall/src/runtime/panic.go:914 +0x21f
github.com/voedger/voedger/pkg/vvm.buildSchemasASTs({0x1664ed8, 0xc007fc0a38}, {0x165abf0?, 0xc0081d23c0?})
	c:/workspace/go/voedger/pkg/vvm/impl_builder.go:52 +0xfb
github.com/voedger/voedger/pkg/vvm.VVMAppsBuilder.Build(0x1337ac0?, 0xc0081a7560, {{0x1655d30, 0xc008194d80}, {0x16519c0, 0xc007f19540}, 0xc0081a7560, {0x1651b80, 0xc0081d22c0}, {0x1651900, ...}, ...}, ...)
	c:/workspace/go/voedger/pkg/vvm/impl_builder.go:64 +0x291
github.com/voedger/voedger/pkg/vvm.provideVVMApps(...)
	c:/workspace/go/voedger/pkg/vvm/wire_gen.go:345
github.com/voedger/voedger/pkg/vvm.ProvideCluster({0x16582e8, 0xc007f19450}, 0xc008197a40, 0x0)
	c:/workspace/go/voedger/pkg/vvm/wire_gen.go:99 +0x9a8
github.com/voedger/voedger/pkg/vvm.ProvideVVM(0xc008197a40, 0x1?)
	c:/workspace/go/voedger/pkg/vvm/wire_gen.go:187 +0x20a
github.com/voedger/voedger/pkg/vit.newVit(0xc007b74ea0, 0xc007d58b00, 0x0)
	c:/workspace/go/voedger/pkg/vit/impl.go:101 +0x306
github.com/voedger/voedger/pkg/vit.NewVIT(0xc007b74ea0, 0xc007d58b00, {0x0, 0x0, 0x100000000000000?})
	c:/workspace/go/voedger/pkg/vit/impl.go:39 +0xe5
github.com/untillpro/airs-bp3/packages/air/it.TestBasicUsage_PaymentMethodsManagement(0xc007b74ea0)
	c:/workspace/go/airs-bp3/packages/air/it/impl_untillpayments_test.go:834 +0xcab
testing.tRunner(0xc007b74ea0, 0x159bde0)
	c:/workspace/go/goinstall/src/testing/testing.go:1595 +0xff
created by testing.(*T).Run in goroutine 1
	c:/workspace/go/goinstall/src/testing/testing.go:1648 +0x3ad
FAIL	github.com/untillpro/airs-bp3/packages/air/it	2.350s
```

--- a/pkg/parser/impl_build.go
+++ b/pkg/parser/impl_build.go
@@ -264,7 +264,7 @@ func (c *buildContext) views() error {
 				for _, ref := range f.RefDocs {
 					if err := resolveInCtx(ref, ictx,
 						func(tbl *TableStmt, pkg *PackageSchemaAST) error {
-							if e := c.checkReference(ref, pkg, tbl, ictx); e != nil {
+							if e := c.checkReference(ref, pkg, tbl); e != nil {
 								return e
 							}
 							refs = append(refs, appdef.NewQName(string(pkg.Name), string(ref.Name)))
@@ -480,7 +480,7 @@ func (c *buildContext) addFieldRefToDef(refField *RefFieldExpr, ictx *iterateCtx
 	errors := false
 	for i := range refField.RefDocs {
 		err := resolveInCtx(refField.RefDocs[i], ictx, func(tbl *TableStmt, pkg *PackageSchemaAST) error {
-			if e := c.checkReference(refField.RefDocs[i], pkg, tbl, ictx); e != nil {
+			if e := c.checkReference(refField.RefDocs[i], pkg, tbl); e != nil {
 				return e
 			}
 			refs = append(refs, appdef.NewQName(string(pkg.Name), string(refField.RefDocs[i].Name)))
@@ -732,7 +732,7 @@ func (c *buildContext) defCtx() *defBuildContext {
 	return &c.defs[len(c.defs)-1]
 }
 
-func (c *buildContext) checkReference(refTable DefQName, pkg *PackageSchemaAST, table *TableStmt, ictx *iterateCtx) error {
+func (c *buildContext) checkReference(refTable DefQName, pkg *PackageSchemaAST, table *TableStmt) error {
 	if refTable.Package == "" {
 		refTable.Package = Ident(pkg.Name)
 	}

--- a/pkg/parser/impl_build.go
+++ b/pkg/parser/impl_build.go
@@ -715,7 +715,7 @@ func (c *buildContext) isExists(qname appdef.QName, kind appdef.TypeKind) (exist
 	}
 }
 
-func (c *buildContext) fundSchemaByPkg(pkg string) *PackageSchemaAST {
+func (c *buildContext) findSchemaByPkg(pkg string) *PackageSchemaAST {
 	for _, ast := range c.app.Packages {
 		if ast.Name == pkg {
 			return ast
@@ -738,7 +738,15 @@ func (c *buildContext) checkReference(refTable DefQName, pkg *PackageSchemaAST, 
 	}
 	refTableType := c.builder.TypeByName(appdef.NewQName(string(refTable.Package), string(refTable.Name)))
 	if refTableType == nil {
-		c.table(c.fundSchemaByPkg(string(refTable.Package)), table, ictx)
+		tableSchema := c.findSchemaByPkg(string(refTable.Package))
+		tableCtx := &iterateCtx{
+			basicContext: &c.basicContext,
+			collection:   tableSchema.Ast,
+			pkg:          tableSchema,
+			parent:       nil,
+		}
+
+		c.table(tableSchema, table, tableCtx)
 		refTableType = c.builder.TypeByName(appdef.NewQName(string(refTable.Package), string(refTable.Name)))
 	}
 

